subsys/fs/fs_impl.h

```diff
@@ -28,6 +29,38 @@ extern "C" {
 const char *fs_impl_strip_prefix(const char *path,
 				 const struct fs_mount_t *mp);
 
+/**
+ * @brief Searches for file system type FS_* identifier, as defined in fs/fs.h,
+ *        compatible with string representation of type.
```

Comment:
This should be more clear that the returned value is the globally unique type identifier (e.g. `FS_FATFS`) provided in a call to `fs_register()` to register support for a file system that claims to be compatible with the requested type name.

Also this can be documented as a non-negative value is a valid index, and `-ENOENT` (or another meaningful error) is returned if no compatible registered file system is found.  Then we don't need `FS_UNKNOWN` as a named value; the failure is a normal error value.